gemviz23/demo/resultwindow.py

```diff
@@ -109,6 +117,39 @@ def _get_uidList(self):
         gen = cat._keys_slice(start, end, ascending)
         return list(gen)  # FIXME: fails here with big catalogs, see issue #51
 
+    def get_md(self, run, doc, key, default=""):
+        """Return a key's value from a document in the run."""
+        obj = run.metadata.get(doc)
+        if obj is None:
+            return default
+        return obj.get(key, default)
+
+    def get_run_detectors(self, run):
```

Comment:
since get_run_detectors, get_run_positioners and get_run_stream_names are basically all the same, shouldn't we have a single function to handle all the items that are lists of strings?